util/throttle.cc

```diff
@@ -283,10 +298,20 @@ void ThrottleShutdown()
     if (gThrottleRunning)
     {
         gThrottleRunning=false;
-        pthread_mutex_lock(&gThrottleMutex);
-        pthread_cond_signal(&gThrottleCond);
-        pthread_mutex_unlock(&gThrottleMutex);
+
+        // lock gThrottleMutex so that we can signal gThrottleCond
+        {
+            MutexLock lock(gThrottleMutex);
+            gThrottleCond->Signal();
+        } // unlock gThrottleMutex
+
         pthread_join(gThrottleThreadId, NULL);
+
+        delete gThrottleCond;
```

Comment:
Problem:  see util/env_posix.cc function Env::Shutdown().  The throttle object is deleted before the compaction threads.  So routine SetThrottleWriteRate() could still be called after the Condition variable and mutex get deleted / set to NULL.  I feel a segfault.

The question becomes whether or not the Throttle could be deleted after the threads shutdown.  Thinking on that right now.